src/Storages/StoragePostgreSQL.cpp

```diff
@@ -376,53 +384,78 @@ SinkToStoragePtr StoragePostgreSQL::write(
 }
 
 
-void registerStoragePostgreSQL(StorageFactory & factory)
+StoragePostgreSQLConfiguration StoragePostgreSQL::getConfiguration(ASTs engine_args, ContextPtr context)
 {
-    factory.registerStorage("PostgreSQL", [](const StorageFactory::Arguments & args)
-    {
-        ASTs & engine_args = args.engine_args;
+    auto [common_configuration, storage_specific_args, with_named_collection] = getExternalDataSourceConfiguration(engine_args, context);
+    StoragePostgreSQLConfiguration configuration(common_configuration);
 
+    if (with_named_collection)
+    {
+        configuration.addresses = {std::make_pair(configuration.host, configuration.port)};
+        for (const auto & [arg_name, arg_value] : storage_specific_args)
+        {
+            if (arg_name == "on_conflict")
+                configuration.on_conflict = arg_value.safeGet<String>();
+            else
+                throw Exception(ErrorCodes::BAD_ARGUMENTS,
```

Comment:
The point is that we cannot do it this way because, for example, the same `StorageURLConfiguration` can have different non common arguments as a table function and as a table engine. The same goes for `StoragePostgreSQLConfiguration`, it has different non common argumetns for database engine, for table engine, for materialized table engine.

I think that it is ok to process configuration commonly, but key-value arguments are processed for each engine separately, becuase they differ from engine type.